## Task 9 — Customer Distribution by Subscription Plan

### Business Question

Which subscription plans have the largest number of customers?

### Analysis

Group customers by subscription plan, calculate the number of customers for each plan, and sort the results in descending order by customer count.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
subscriptions_users_amount = starlink.groupby("plan_name").agg(
    users_amount = ("unique_id","count")
).sort_values(
    "users_amount",
    ascending = False
)
print(subscriptions_users_amount)

             users_amount
plan_name                
Residential          5959
Roam                 2078
Business             1019
Mini                  944


### Result

The Residential plan has the largest customer base with 5,959 customers, followed by Roam with 2,078 customers. Business has 1,019 customers, while Mini has the smallest customer base with 944 customers.

### Business Insight

The Residential plan has a significantly larger customer base than the other subscription plans and represents the main customer segment. Roam is the second-largest segment, while Business and Mini have substantially smaller customer bases.